module/VuFind/src/VuFind/Db/Table/ResourceTags.php

```diff
@@ -237,10 +237,16 @@ public function destroyLinks($resource, $user, $list = null, $tag = null)
         $callback = function ($select) use ($resource, $user, $list, $tag) {
             $select->where->equalTo('user_id', $user);
             if (null !== $resource) {
-                if (!is_array($resource)) {
-                    $resource = [$resource];
+                if ($resource === 'listTag') {
```

Comment:
Would it be better to have a separate method rather than using a special case here? This is one of several things that makes me wonder if we'd be better off having a separate table for this functionality rather than overloading user_resource/resource_tags.

Reply:
I still don't like the use of a magic string here. Would it be better to create separate destroyListLinks / destroyResourceLinks methods? If that's too redundant/inefficient, maybe we could at least replace the hard-coded string with a class constant to make it more clear that it has a very specific meaning.